libobs/util/platform-nix.c

```diff
@@ -298,12 +457,37 @@ int os_unlink(const char *path)
 
 int os_mkdir(const char *path)
 {
-	if (mkdir(path, 0777) == 0)
+	if (mkdir(path, 0755) == 0)
 		return MKDIR_SUCCESS;
 
 	return (errno == EEXIST) ? MKDIR_EXISTS : MKDIR_ERROR;
 }
 
+int os_mkdirs(const char *path, int mode) {
```

Comment:
This wouldn't port to windows.  You cannot set the permissions like this.  It's why os_mkdir does not have this parameter in the first place.